### PR TITLE
fixed system.delete

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

File: lib/system.nim [hunk 1]
```diff
@@ -2134,26 +2134,25 @@ const
 import system/dollars
 export dollars
 
-proc delete*[T](x: var seq[T], i: Natural) {.noSideEffect.} =
+proc delete*[T](x: var seq[T], i: Natural) {.noSideEffect, deprecated: "use sequtils.delete(x, i..i) instead".} =
```
Comment (konsumlamm) [inline lib/system.nim]:
```suggestion
proc delete*[T](x: var seq[T], i: Natural) {.noSideEffect, deprecated: "use `sequtils.delete(x, i..i)` instead".} =
```

Comment (juancarlospaco) [reply lib/system.nim]: I dont know, other `"use X instead"` do not have pseudocode for the arguments anyway, would be better to be consistent. 🤔

Comment (konsumlamm) [reply lib/system.nim]: Eh, most deprecations I found used backticks for the replacement function/expression.

Comment (Araq) [reply lib/system.nim]: > I dont know, other "use X instead" do not have pseudocode for the arguments anyway, would be better to be consistent.

Well we cannot all be so smart and remove useful information for "consistency".

**Comment by juancarlospaco** [comment]
So you can now `add()`, but you can not `delete()` with `system.nim` ?. 🤔

**Comment by konsumlamm** [comment]
> So you can now `add()`, but you can not `delete()` with `system.nim` ?. thinking

I don't see the problem to be honest. `add` is way more common than `delete`, one of the reasons being that it is faster (amortized O(1) vs O(n)).

**Comment by Varriount** [comment]
> 
> 
> > So you can now `add()`, but you can not `delete()` with `system.nim` ?. thinking
> 
> I don't see the problem to be honest. `add` is way more common than `delete`, one of the reasons being that it is faster (amortized O(1) vs O(n)).

Yeah, but it's still a basic operation. Should we move the `set` operators `<=`, `*`, etc. from the system module, just because they aren't used as often as `contains`?

**Comment by Araq** [comment]
This is still a terrible solution: It punishes everybody who used system.delete successfully with correct indexes for years...

Finally found a good compromise, relax.

**Comment by juancarlospaco** [comment]
I did not say "theres a problem".

I just think theres no way you pass this change easily thru CI of packages, I guess people do use `delete`.
🙂

**Comment by Varriount** [comment]
Aside from pointing to an implementation that doesn't have confusing semantics, what benefit does moving `delete` from system to sequtils offer?

**Comment by Araq** [comment]
> what benefit does moving delete from system to sequtils offer?

System.nim should be smaller, it's the one module most expensive to touch, just look at how the code could not call `raiseIndexError2` directly.

**Pull request merged**